ytmusicapi/mixins/browsing.py

```diff
@@ -44,10 +46,31 @@ def search(self, query: str, filter: str = None) -> List[Dict]:
               }
             ]
         """
-        body = {'query': query}
-        endpoint = 'search'
         search_results = []
-        filters = ['albums', 'artists', 'playlists', 'songs', 'videos', 'uploads']
+        continuationToken = None
+        try:
+            while len(search_results) < maxResults:
+                result = self._base_search(q=query,
+                                           filter=filter,
+                                           continuationToken=continuationToken)
+                search_results.extend(result["search_results"])
+                continuationToken = result["nextPageToken"]
+                if continuationToken is None:
+                    break
```

Comment:
This logic is already in get_continuations

Reply:
Noted.